x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecker.java

```diff
@@ -115,6 +131,17 @@ private DeprecationIssue oldIndicesCheck(
                     false,
                     Map.of("reindex_required", true, "transform_ids", transforms)
                 );
+            } else if (cldrIncompatibleFieldMappings.isEmpty() == false) {
+                return new DeprecationIssue(
+                    DeprecationIssue.Level.CRITICAL,
+                    "Field mappings with incompatible date format patterns in old index",
+                    "https://www.elastic.co/blog/locale-changes-elasticsearch-8-16-jdk-23",
+                    "This index has a compatibility version < 8.0 and field mappings with format patterns that are incompatible "
```

Comment:
The mappings aren't incompatible with 9.0, it's just they can't be reindexed b/c of CLDR. Perhaps something like:
`The index was created before 8.0 and contains mappings that must be reindexed due to locale changes in 8.16+. Manual reindexing is required.`

Reply:
I've gone with your suggested wording.